quinn-proto/src/transport_parameters.rs

```diff
@@ -476,6 +475,43 @@ fn decode_cid(len: usize, value: &mut Option<ConnectionId>, r: &mut impl Buf) ->
     Ok(())
 }
 
+// Write transport parameter with up to 16 bytes of random payload and ID from reserved values.
+fn write_random_grease_reserved_parameter<W: BufMut, R: RngCore>(w: &mut W, rng: &mut R) {
+    // See: https://datatracker.ietf.org/doc/html/rfc9000#section-22.3
+    // Inspired by quic-go & quiche implementation:
+    // [1] https://github.com/quic-go/quic-go/blob/3e0a67b2476e1819752f04d75968de042b197b56/internal/wire/transport_parameters.go#L338-L344
+    // [2] https://github.com/google/quiche/blob/cb1090b20c40e2f0815107857324e99acf6ec567/quiche/quic/core/crypto/transport_parameters.cc#L843-L860
+
+    // The randomly generated identifier of the form `31 * N + 27`;
+    let id = {
+        // To generate random ID it first necessary to pick random `N` such that `31 * N + 27 < 2^62`,
+        // It is achieved by picking random number from range [0; 2^62 - 27) and dividing it by 31.
```

Comment:
This describes the math being done, but we can already see that from the actual code. Please instead document *why* this is necessary. Presumably the idea is to satisfy the RFC's definition of IDs that must never be assigned? If so, explain that, and cite https://www.rfc-editor.org/rfc/rfc9000.html#name-quic-transport-parameters-r.